db/articles.js

```diff
@@ -0,0 +1,76 @@
+let articles = [];
+
+function all() {
+  return articles;
+}
+
+function createURLTitle(title) {
+  let newTitle = encodeURIComponent(title);
+  return newTitle;
+}
+
+function checkArticleExists(inputTitle){
+  for(let i=0; i<articles.length; i++){
+    let current = articles[i];
+    if(current.title === inputTitle){
+      return true
+    }
+  }
+  return false;
+}
+
+function findArticleByTitle(inputTitle) {
+  let article = {};
+  for(let i=0; i<articles.length; i++){
+    let current = articles[i];
+    if(current.title === inputTitle){
+      article = current
+    }
+  }
+
+  if(article === {}){
```

Comment:
Rather than do the check like this of setting an empty object and comparing it that way it's better to leave the variable as `undefined` at time of creation and use truthiness to evaluate your assignment condition.

```
let article;
.
.
.
if (!article) {
  return false;
}
```